forge-ai/src/main/java/forge/ai/SpellAbilityAi.java

```diff
@@ -81,23 +86,23 @@ protected boolean canPlayWithoutRestrict(final Player ai, final SpellAbility sa)
             final boolean alwaysOnDiscard = "AlwaysOnDiscard".equals(logic) && ai.getGame().getPhaseHandler().is(PhaseType.END_OF_TURN, ai)
                     && !ai.isUnlimitedHandSize() && ai.getCardsIn(ZoneType.Hand).size() > ai.getMaxHandSize();
             if (!checkAiLogic(ai, sa, logic)) {
-                return false;
+                return new AiAbilityDecision(0, AiPlayDecision.CantPlayAi);
             }
             if (!alwaysOnDiscard && !checkPhaseRestrictions(ai, sa, ai.getGame().getPhaseHandler(), logic)) {
-                return false;
+                return new AiAbilityDecision(0, AiPlayDecision.MissingPhaseRestrictions);
             }
         } else if (!checkPhaseRestrictions(ai, sa, ai.getGame().getPhaseHandler())) {
-            return false;
+            return new AiAbilityDecision(0, AiPlayDecision.MissingPhaseRestrictions);
         }
 
         if (!checkApiLogic(ai, sa)) {
```

Comment:
can you also refactor this one?
it's actually used by more effects

I'll gladly update the ones still overriding `canPlayAI` in a follow-up because they really should just use one of the more tailored entrypoints